src/goto-instrument/contracts/contracts.cpp

```diff
@@ -686,106 +685,61 @@ void code_contractst::instrument_assign_statement(
 }
 
 void code_contractst::instrument_call_statement(
-  goto_programt::instructionst::iterator &instruction_iterator,
-  goto_programt &program,
-  assigns_clauset &assigns_clause)
+  goto_programt::instructionst::iterator &instruction_it,
+  const irep_idt &function,
+  goto_programt &body,
+  assigns_clauset &assigns)
 {
   INVARIANT(
-    instruction_iterator->is_function_call(),
+    instruction_it->is_function_call(),
     "The first argument of instrument_call_statement should always be "
     "a function call");
 
-  irep_idt called_name;
-  if(instruction_iterator->call_function().id() == ID_dereference)
+  irep_idt called_function;
+  if(instruction_it->call_function().id() == ID_dereference)
   {
-    called_name =
+    called_function =
```

Comment:
Surely we want a test for this.

Reply:
This is to handle function pointers. We should perhaps abort and report error message in this case. Correct?

Reply:
This is showing up as uncovered, I don't think we should have uncovered code in a system that must be trusted. But, if this is for function pointers and we aren't yet correctly instrumenting function pointers then let's abort here with an error message.

Reply:
We don't need this case anymore. Inlining takes care of that problem, since all function calls (including the ones with function pointers) are inlined during enforcement.